frontend/src/registration/components/DomainInputPage/DomainInputPage.tsx

```diff
@@ -53,16 +58,45 @@ export class DomainInputPage extends React.PureComponent<Props, State> {
     }
   };
 
+  private handleExternalDomainChange = (newDomain: string) => {
+    this.setState({
+      externalDomain: newDomain
+    });
+    // Clean up error feedback if any
+    if (this.props.registrationFeedback.subdomain) {
+      this.props.clearErrorMessage('externalDomain');
+    }
+  };
+
   private submitDomain = () => {
-    this.props.performValidationAndStore(
-      {
-        subdomain: this.state.subdomain
-      },
-      RegistrationSteps.INSTANCE
-    );
+    if (this.state.domainIsExternal) {
+      let subdomain = this.state.externalDomain.replace('.', '');
+      // Append random digits to internal domain to avoid conflicts
+      subdomain = `${subdomain}-${Math.random()
+        .toString(36)
+        .substr(2, 4)}`;
+      this.props.performValidationAndStore(
+        {
+          subdomain,
+          externalDomain: this.state.externalDomain
+        },
+        RegistrationSteps.CUSTOM_DOMAIN
+      );
+    } else {
+      this.props.performValidationAndStore(
+        {
+          subdomain: this.state.subdomain
+        },
+        RegistrationSteps.INSTANCE
+      );
+    }
   };
```

Comment:
I'm wondering if it would be better to just ask the user for an internal domain as well. It is used in a lot of places, for instance to generate the database names etc so might be best to give users an option.

Reply:
I think we should avoid this for now. It won't make much sense for a user that's registering for a external domain to be asked for an internal one too.

Reply:
We're already discussing removing the instance extra email from the registration form because users get confused with having to activate 2 emails.